WordPress/gutenberg
# Refactor editEntityRecord, undo, and redo to be thunks instead of generators

Builds on top of the thunks support added in #27276 and refactors just the parts of core-data required to make the `editEntityRecord`, `undo`, and `redo` work (see #28389 from @jsnajdr).

**Test plan:**

* Confirm all the tests are green.
* Go to the post editor, edit a post, add a Blog Title block and edit it as well, save everything and confirm it worked as expected.
* Try undo-ing and redo-ing adding a block, typing a few lines of text, pasting content.
* Go to the widgets editor, edit a few widgets, click Update, confirm it worked as expected.
* Try anything else edit-related that comes to your mind.
